worker/sshserver/listener_test.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package sshserver_test
+
+import (
+	"time"
+
+	"github.com/juju/testing"
+	"google.golang.org/grpc/test/bufconn"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/worker/sshserver"
+)
+
+type listenerSuite struct {
+	testing.IsolationSuite
+}
+
+var _ = gc.Suite(&listenerSuite{})
+
+func (s *listenerSuite) TestAcceptOnceListener(c *gc.C) {
+	listener := bufconn.Listen(8 * 1024)
+
+	acceptOnceListener, closeAllowed := sshserver.NewSSHServerListener(listener)
+	c.Assert(acceptOnceListener, gc.NotNil)
+
+	done := make(chan bool, 1)
```

Comment:
Bool tells you nothing here, just use a struct{}.

```suggestion
	done := make(chan struct{})
```